selfservice/strategy/oidc/provider_generic_oidc.go

```diff
@@ -97,16 +109,31 @@ func (g *ProviderGenericOIDC) verifyAndDecodeClaimsWithProvider(ctx context.Cont
 	return &claims, nil
 }
 
+func verifyAudience(received []string, expected []string) error {
+	for _, r := range received {
+		for _, e := range expected {
+			if r == e {
+				return nil
+			}
+		}
+	}
+	return fmt.Errorf("oidc: audience not valid: %v", received)
```

Comment:
To add a stack trace we use `pkg/errors`:

```suggestion
	return errors.Errorf("oidc: audience not valid: %v", received)
```